### ISSUE TITLE
I am getting not an valid XML Type for Sprindata repository java class.

### ISSUE THREAD

**Issue by rishikv** [stoicflame/enunciate#308]
I am using java 7 and trying to run enunciate and stumbled upon this error. Any insights or how to fix this issue is greatly appreciate, i spent more than half day how to fix this.

```
com.webcohesion.enunciate.EnunciateException: freemarker.core._TemplateModelException: An error has occurred when reading existing sub-variable "ref"; see cause exception! The type of the containing value was: extended_hash+string (com.webcohesion.enunciate.modules.jaxb.model.Element wrapped into f.e.b.StringModel)

----
FTL stack trace ("~" means nesting-related):
	- Failed at: #if ((!accessor.elementRefs!false) &&...  [in template "jar:file:/Users/rishikeshkollipara/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-c-xml-client/2.2.0/e55098a2bcb49b40b66e53764ea6c9c7bc7df079/enunciate-c-xml-client-2.2.0.jar!/com/webcohesion/enunciate/modules/c_client/client-complex-type.fmt" in macro "writeComplexTypeHeader" at line 16, column 5]
- Reached through: @libComplexType.writeComplexTypeHeade...  [in template "jar:file:/Users/rishikeshkollipara/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-c-xml-client/2.2.0/e55098a2bcb49b40b66e53764ea6c9c7bc7df079/enunciate-c-xml-client-2.2.0.jar!/com/webcohesion/enunciate/modules/c_client/api.fmt" at line 31, column 13]
- Reached through: @file name=cFileName charset="utf-8"  [in template "jar:file:/Users/rishikeshkollipara/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-c-xml-client/2.2.0/e55098a2bcb49b40b66e53764ea6c9c7bc7df079/enunciate-c-xml-client-2.2.0.jar!/com/webcohesion/enunciate/modules/c_client/api.fmt" at line 14, column 1]
----

Caused by: com.webcohesion.enunciate.EnunciateException: com.mycompany.springdata.MyStateRepository is not recognized as an XML type.
	at com.webcohesion.enunciate.modules.jaxb.model.types.XmlTypeVisitor.defaultAction(XmlTypeVisitor.java:41)
	at com.webcohesion.enunciate.modules.jaxb.model.types.XmlTypeVisitor.defaultAction(XmlTypeVisitor.java:37)
	at com.webcohesion.enunciate.modules.jaxb.model.types.XmlTypeVisitor.visitDeclared(XmlTypeVisitor.java:101)
	at com.webcohesion.enunciate.modules.jaxb.model.types.XmlTypeVisitor.visitDeclared(XmlTypeVisitor.java:37)
	at com.webcohesion.enunciate.javac.decorations.type.DecoratedDeclaredType.accept(DecoratedDeclaredType.java:83)
	at com.webcohesion.enunciate.modules.jaxb.model.types.XmlTypeFactory.getXmlType(XmlTypeFactory.java:161)
	at com.webcohesion.enunciate.modules.jaxb.model.Accessor.getBaseType(Accessor.java:181)
	at com.webcohesion.enunciate.modules.jaxb.model.Element.getBaseType(Element.java:234)
	at com.webcohesion.enunciate.modules.jaxb.model.Element.getRef(Element.java:165)
	... 86 more
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class
	at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:101)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:81)
	... 30 more
```

**Comment by stoicflame** [comment]
Looks like a bug. Thanks for the report.

**Comment by rishikv** [comment]
as a workaround for now if i disabled jaxb and jaxws.

**Comment by stoicflame** [comment]
I believe this happens to be a duplicate of #270 and has been fixed at f23e33e.

**Comment by stoicflame** [comment]
[Enunciate 2.3 has been released](https://github.com/stoicflame/enunciate/releases/tag/v2.3.0).
